### PR TITLE
Feature/adding swagger documentation

### PR DESCRIPTION
**Pull request by varunkumar139**

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppslaunchpadauth/resource/TokenController.kt [hunk 1]
```diff
@@ -28,13 +32,20 @@ class TokenController(
...
-    @RequestParam("grant_type", required = false) grant: String?,
+    @Parameter(required = true, description = "authorization_code or refresh_token")
+    @RequestParam("grant_type") grant: String,
+    @Parameter(required = false, description = "redirect uri")
     @RequestParam("redirect_uri", required = false) redirectUri: URI?,
+    @Parameter(required = false, description = "refresh token in jwt")
     @RequestParam("refresh_token", required = false) refreshToken: String?,
+    @Parameter(required = false, description = "nonce")
     @RequestParam(required = false) nonce: String?,
+    @Parameter(required = true, description = "Basic authorization header with client id as username and client secret as password")
```
Comment (tosin-ogunrinde) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/hmppslaunchpadauth/resource/TokenController.kt:48]: description = "HTTP  Basic authentication header with the client_id as the username and client secret as the password.", 
example = "Basic MGRkM..."

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppslaunchpadauth/dto/Token.kt [hunk 2]
```diff
@@ -1,20 +1,26 @@
...
 
+  @Schema(required = true, description = "Access token", example = "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJzdWIiOiIxMjM0NTY3ODkwIiwibmFtZSI6IkpvaG4gRG9lIiwiaWF0IjoxNTE2MjM5MDIyfQ.SflKxwRJSMeKKF2QT4fwpMeJf36POk6yJV_adQssw5c")
   @JsonProperty("access_token")
   val accessToken: String,
 
+  @Schema(required = true, description = "Refresh token", example = "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJzdWIiOiIxMjM0NTY3ODkwIiwibmFtZSI6IkpvaG4gRG9lIiwiaWF0IjoxNTE2MjM5MDIyfQ.SflKxwRJSMeKKF2QT4fwpMeJf36POk6yJV_adQssw5c")
   @JsonProperty("refresh_token")
   val refreshToken: String,
 
+  @Schema(required = true, description = "Token type is always bearer", example = "Bearer")
```
Comment (tosin-ogunrinde) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/hmppslaunchpadauth/dto/Token.kt:19]: description = "Always set to `Bearer` to indicate that the token is a bearer token."

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppslaunchpadauth/resource/UserApprovedClientController.kt [hunk 3]
```diff
@@ -28,17 +31,23 @@ import java.util.*
...
   fun getUserApprovedClients(
+    @Parameter(required = true, description = "user id")
     @PathVariable("user-id") userId: String,
+    @Parameter(required = false, description = "page number, default value 1")
     @RequestParam("page", required = false) page: Long?,
+    @Parameter(required = false, description = "number of record in a page, max value is 20")
     @RequestParam("size", required = false) size: Long?,
+    @Parameter(required = true, description = "jwt access token")
     @RequestHeader(HttpHeaders.AUTHORIZATION, required = true) authorization: String,
   ): ResponseEntity<PagedResult<UserApprovedClientDto>> {
```
[the remark this answers is not in the document]
Comment (tosin-ogunrinde) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/hmppslaunchpadauth/resource/UserApprovedClientController.kt:52]: Can PagedResultUserApprovedClientDto be exposed as  PagedResultUserApprovedClient, and UserApprovedClientDto to UserApprovedClient?

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppslaunchpadauth/resource/AuthController.kt [hunk 4]
```diff
@@ -28,24 +35,52 @@ class AuthController(private var ssoLoginService: SsoLogInService) {
...
+        responseCode = "302",
+        description = "Redirect to url provided in redirect uri",
+      ),
+      ApiResponse(
+        responseCode = "400",
+        description = "Bad request",
+      ),
+      ApiResponse(
+        responseCode = "403",
+        description = "Access Denied",
```
[the remark this answers is not in the document]
Comment (varunkumar139) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/hmppslaunchpadauth/resource/AuthController.kt:55]: I think it is OK to have, I have also added error schema for response code like 400, 403, 401

**Pull request merged**